vm/src/obj/objdict.rs

```diff
@@ -247,6 +234,54 @@ impl ItemProtocol for PyDictRef {
     }
 }
 
+// Implement IntoIterator so that we can easily iterate dictionaries from rust code.
+impl IntoIterator for PyDictRef {
+    type Item = (PyObjectRef, PyObjectRef);
+    type IntoIter = DictIterator;
+
+    fn into_iter(self) -> Self::IntoIter {
+        DictIterator::new(self)
+    }
+}
+
+impl IntoIterator for &PyDictRef {
+    type Item = (PyObjectRef, PyObjectRef);
+    type IntoIter = DictIterator;
+
+    fn into_iter(self) -> Self::IntoIter {
+        DictIterator::new(self.clone())
+    }
+}
+
+pub struct DictIterator {
+    dict: PyDictRef,
+    position: Cell<usize>,
```

Comment:
Why the `Cell`? `Iterator::next` takes `&mut self`

Reply:
Originally just because that's how I had to do the python ones. It works here though. I have to get rid of the ```.map``` to avoid ```closure requires unique access to `self` but it is already borrowed```.

(I'm not sure how I feel about ```.map```.)